### PR TITLE
Change default orbit yaw behaviour

### PR DESCRIPTION
**Pull request by Claudio-Chies**
Change of default orbit yaw behaviour.
-----------
Associated PR: 

- [x] https://github.com/PX4/PX4-Autopilot/pull/23358
- [x] https://github.com/mavlink/mavlink/pull/2132

Description
-----------
when a vehicle type, for which the orientation is relevant, performs an orbit maneuver, e.g a Tricopter, the orientation during the orbit had to be hardcoded. With this PR i change [ORBIT_YAW_BEHAVIOUR](https://mavlink.io/en/messages/common.html#ORBIT_YAW_BEHAVIOUR) from [ORBIT_YAW_BEHAVIOUR_HOLD_FRONT_TO_CIRCLE_CENTER](https://mavlink.io/en/messages/common.html#ORBIT_YAW_BEHAVIOUR_HOLD_FRONT_TO_CIRCLE_CENTER) to  [ORBIT_YAW_BEHAVIOUR_UNCHANGED](https://mavlink.io/en/messages/common.html#ORBIT_YAW_BEHAVIOUR_UNCHANGED) .
if the user uses a PX4 version where the mentioned PR is not active, it remains ORBIT_YAW_BEHAVIOUR_HOLD_FRONT_TO_CIRCLE_CENTER as this is what is set in the constructor of MAV_CMD_DO_ORBIT. so backwards compatibility is given. if The user used a PX4 version with the mentioned PR, it allows the user to set the orbit yaw behaviour based on a vehicle parameter.

Test Steps
-----------
Tested in SITL

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by Claudio-Chies** [comment]
> What about ArduPilot? How will it respond to this change?

from what i can see the MAV_CMD_DO_ORBIT is not implemented in Ardupilot. so nothing should break.
but i havent used/worked with ardupilot.

**Comment by DonLakeFlyer** [comment]
@Claudio-Chies Let me know when the PX4 side gets merged

**Comment by Claudio-Chies** [comment]
Will do, still have an open discussions regarding the mavlink change

**Comment by Claudio-Chies** [comment]
adapted the change requested in https://github.com/mavlink/mavlink/pull/2132
Tested with and without https://github.com/PX4/PX4-Autopilot/pull/23358, both cases work as expected.

**Comment by DonLakeFlyer** [comment]
Also this pull needs to be updated to use ORBIT_YAW_BEHAVIOUR_DEFAULT instead of a magic number once mavlink is updated.

**Comment by Claudio-Chies** [comment]
> Also this pull needs to be updated to use ORBIT_YAW_BEHAVIOUR_DEFAULT instead of a magic number once mavlink is updated.

i adapted the code to use the enum, but something seems to not work.
mabye you can give me a pointer.
- the mavlink git submodule is set to use current upstream master in detached mode. and i see `ORBIT_YAW_BEHAVIOUR_UNCHANGED` in my `common.xml`
- when building i get:
`.../qgroundcontrol/src/Vehicle/Vehicle.cc:2176:40: error: ‘ORBIT_YAW_BEHAVIOUR_UNCHANGED’ was not declared in this scope; did you mean ‘ORBIT_YAW_BEHAVIOUR_UNCONTROLLED’?
 2176 |                     static_cast<float>(ORBIT_YAW_BEHAVIOUR_UNCHANGED),       // Use current or vehicle default yaw behavior
`
do i have to build/integrate mavlink first before building QGC? i cant seem to find anything in the documentation.

File: src/VideoManager/VideoManager.cc
Comment (Claudio-Chies) [inline src/VideoManager/VideoManager.cc:1]: i had to make these changes to update the mavlink version, as the most current version changed the enum
see https://github.com/mavlink/mavlink/pull/2127 but was unable to test it

**Comment by sfuhrer** [comment]
@DonLakeFlyer what's required to get this in?

**Pull request merged**